src/include/event_config.h

```diff
@@ -126,7 +127,7 @@ extern GHashTable *g_event_config_list;   // for iterating through entire list o
 GList *export_event_config(const char *event_name);
 void unexport_event_config(GList *env_list);
 
-GHashTable *validate_event(const char *event_name);
+GList *get_options_with_err_msg(const char *event_name);
```

Comment:
I am sorry but I do not think this is a good idea. The function has an unexpected side effect. You cannot call it twice - if you do you get a memory leak - and the `eo_error_msg` member become nonsense when you set the value.

The function `validate_event` was supposed to return a list of options with invalid values (no value is an invalid value for mandatory options). The original author used `GHashTable` as the easiest way of passing the pairs. What about to create a new structure, let's say `struct invalid_options { const char *opt_name; char *error; };`, and return `GList *` of these structures.